hack/generated/controllers/suite_test.go

```diff
@@ -124,22 +124,45 @@ func teardown() error {
 	return nil
 }
 
-func TestMain(m *testing.M) {
+type controllerTest struct {
+	name string
+	test func(ctx ControllerTestContext, t *testing.T)
+}
 
-	err := setup()
-	if err != nil {
-		panic(err)
+func runTests(t *testing.T, tests []controllerTest) {
+	ctx, setupErr := setup()
```

Comment:
Are we worried about `ctx context.Context` and `ctx ControllerTestContext` confusion here?

Should we call the `ControllerTestContext` something else?